product_status/models/product.py

```diff
@@ -32,37 +40,61 @@ class ProductProduct(models.Model):
         string="New until",
         help="New product, and you want to warn your client for a certain time",
     )
+    tmpl_end_of_life_date = fields.Date(
+        string="Template End-of-life",
+        related="product_tmpl_id.end_of_life_date",
+    )
+    tmpl_discontinued_until = fields.Date(
+        string="Template Discontinued until",
+        related="product_tmpl_id.discontinued_until",
+    )
+    tmpl_new_until = fields.Date(
+        string="Template New until",
+        related="product_tmpl_id.new_until",
+    )
+    has_status_date = fields.Boolean(
+        compute="_compute_has_status_date",
+        store=False,
+    )
 
     @api.onchange("end_of_life_date")
     def _onchange_end_of_life_date(self):
         for rec in self:
-            if rec.end_of_life_date:
-                rec.update({"new_until": False, "discontinued_until": False})
+            self.product_tmpl_id._update_dates_of_states(rec, "end_of_life_date")
```

Comment:
A loop is not needed in onchanges, but if you keep it, fix this `self.product_tmpl_id` -> `rec.product_tmpl_id`

Reply:
Thanks for your input. Actually the `_update_dates_of_states` method does not use self. The actual record need to be updated is passed to the method as `rec`. I have just add `@api.model` decorator to the method.

Reply:
not really intuitive :sweat_smile: but never mind :wink: